examples/arkode/CXX_parallel/ark_diffusion_reaction_p.cpp

```diff
@@ -1016,9 +1022,17 @@ static int SetupARK(SUNContext ctx, UserData* udata, N_Vector u,
   }
   else
   {
-    flag = ARKStepSetAdaptivityMethod(*arkode_mem, udata->controller, SUNTRUE,
-                                      SUNFALSE, NULL);
-    if (check_flag(&flag, "ARKStepSetAdaptivityMethod", 1)) return 1;
+    SUNControl Ctrl = NULL;
+    switch (udata->controller) {
+    case (ARK_ADAPT_PID):      Ctrl = SUNControlPID(ctx);     break;
+    case (ARK_ADAPT_PI):       Ctrl = SUNControlPI(ctx);      break;
+    case (ARK_ADAPT_I):        Ctrl = SUNControlI(ctx);       break;
+    case (ARK_ADAPT_EXP_GUS):  Ctrl = SUNControlExpGus(ctx);  break;
+    case (ARK_ADAPT_IMP_GUS):  Ctrl = SUNControlImpGus(ctx);  break;
+    case (ARK_ADAPT_IMEX_GUS): Ctrl = SUNControlImExGus(ctx); break;
```

Comment:
> It might be worth deprecating the current `ARKStepSetAdaptivityMethod` signature but changing it later so users can select the native controllers without having to construct an object (unless they want to alter the default parameters).

I second this suggestion.

Reply:
Then let's add this "changing it later..." to our longer-term to-do list, since we certainly cannot change it yet.

Reply:
Also, ARKODE destroys whatever controller and heuristics objects it has in-hand when it is destroyed, thus there is no need to manually destroy it in the examples (if you do, you get a double-free).  If you feel strongly that we should carry around a bool to say whether ARKODE "owns" those objects, and then only frees them if it owned them originally, let me know.

Reply:
I had converted all uses of `ARKStepSetAdaptivityMethod` to the new approach, but I've just reverted the change in `ark_heat1D_adapt.c` to serve as a unit test.  That said, these kind of unit tests should not reside in `examples` since we should strive for those to show "best practices" instead of deprecated functions.